internal/svc/admin_svc.go

```diff
@@ -246,19 +244,19 @@ func (cas *CerbosAdminService) ReloadStore(ctx context.Context, _ *requestv1.Rel
 		return nil, status.Error(codes.Unimplemented, "Configured store is not reloadable")
 	}
 
-	_, err, shared := sfGroup.Do("admin_reload", func() (interface{}, error) {
-		if err := rs.Reload(ctx); err != nil {
-			log.Error("Failed to reload the store", zap.Error(err))
-			return nil, status.Errorf(codes.Internal, "Failed to reload the store")
+	if req.Wait {
+		err := storage.Reload(ctx, log, rs)
+		if err != nil {
+			log.Error("failed to reload the store due to an error: %w", err)
```

Comment:
I think using `%w` to output errors is not correct (I am surprised the linter didn't catch it). Anyway, when logging errors we should do `log.Error("msg", zap.Error(err))` to make the error a component of the structured logging output. Otherwise some errors won't show up properly in log aggregators.